app/src/main/java/at/bitfire/icsdroid/ui/EditCalendarActivity.kt

```diff
@@ -96,60 +99,170 @@ class EditCalendarActivity: AppCompatActivity() {
             }
         }
 
-        // show error message from calling intent, if available
-        if (inState == null)
-            intent.getStringExtra(EXTRA_ERROR_MESSAGE)?.let { error ->
-                AlertFragment.create(error, intent.getSerializableExtra(EXTRA_THROWABLE) as? Throwable)
-                    .show(supportFragmentManager, null)
+        // Whether unsaved changes exist
+        modelsDirty = object : MediatorLiveData<Boolean>() {
```

Comment:
I don't love this initialization method... I get that it works, but making variables that should not be mutable, mutable... I guess you have initialized it here so that the activity is initialized and a context can be provided. If that's the case, we can initialize the `MediatorLiveData` lazily:
```kotlin
private val modelsDirty: MutableLiveData<Boolean> by lazy {
    // Whether unsaved changes exist
    object : MediatorLiveData<Boolean>() {
        init {
            addSource(subscriptionSettingsModel.title) { value = subscriptionDirty() }
            addSource(subscriptionSettingsModel.color) { value = subscriptionDirty() }
            addSource(subscriptionSettingsModel.ignoreAlerts) { value = subscriptionDirty() }
            addSource(subscriptionSettingsModel.defaultAlarmMinutes) { value = subscriptionDirty() }
            addSource(subscriptionSettingsModel.defaultAllDayAlarmMinutes) { value = subscriptionDirty() }
            addSource(credentialsModel.username) { value = credentialDirty() }
            addSource(credentialsModel.password) { value = credentialDirty() }
        }
        fun subscriptionDirty() = initialSubscription?.let {
            !subscriptionSettingsModel.equalsSubscription(it)
        } ?: false
        fun credentialDirty() = initialCredentials?.let {
            !credentialsModel.equalsCredential(it)
        } ?: false
    }
}
```
It's not the best looking syntax, but at least we use `val`.

Reply:
Not sure if its what you say, but one can't access the ViewModel before onCreate() has been called. That's the reason I made it a "lateinit var". You are right however - using "by lazy" works just fine. Thanks, will change :+1:

Reply:
> Not sure if its what you say, but one can't access the ViewModel before onCreate() has been called

Yup, that's what I meant 😉